internal/wrphandlers/xmidt_agent_crud/handler.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2024 Comcast Cable Communications Management, LLC
+// SPDX-License-Identifier: Apache-2.0
+
+package xmidt_agent_crud
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/xmidt-org/wrp-go/v3"
+	"github.com/xmidt-org/xmidt-agent/internal/loglevel"
+	"github.com/xmidt-org/xmidt-agent/internal/wrpkit"
+)
+
+const DefaultLogLevelChangeMinutes = 30
+
+const (
+	Create   = 5
+	Retrieve = 6
+	Update   = 7
+	Delete   = 8
+)
+
+type Handler struct {
+	egress   wrpkit.Handler
+	source   string
+	logLevel loglevel.LogLevel
+}
+
+// New creates a new instance of the Handler struct.  The parameter egress is
+// the handler that will be called to send the response.  The parameter source is the source to use in
+// the response message. This handler handles crud messages specifically for xmdit-agent, only.
+func New(egress wrpkit.Handler, source string, logLevel loglevel.LogLevel) (*Handler, error) {
+
+	h := Handler{
+		egress:   egress,
+		source:   source,
+		logLevel: logLevel,
+	}
+
+	return &h, nil
+}
+
+func (h *Handler) HandleWrp(msg wrp.Message) error {
+	payload := make(map[string]interface{})
+
+	err := json.Unmarshal(msg.Payload, &payload)
+	if err != nil { // do we still want to return a response here?
+		return err // this would be sent to the eventor error handler
+	}
+
+	var payloadResponse []byte
+	statusCode := int64(http.StatusOK)
+
+	switch msg.Type {
+	case Update:
+		statusCode, _ = h.update(msg.Path, payload)
+		// the above error needs to be logged
```

Comment:
Set the payload response to be the err if present, so it is sent to the original caller.